pkg/controller/agent/reconcile.go

```diff
@@ -36,25 +37,51 @@ func reconcilePodVehicle(params Params, podTemplate corev1.PodTemplateSpec) (*re
 	spec := params.Agent.Spec
 	name := Name(params.Agent.Name)
 
-	var toDelete client.Object
+	var toDelete []client.Object
 	var reconciliationFunc func(params ReconciliationParams) (int32, int32, error)
 	switch {
 	case spec.DaemonSet != nil:
 		reconciliationFunc = reconcileDaemonSet
-		toDelete = &v1.Deployment{
+		toDelete = append(toDelete, &v1.Deployment{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      name,
 				Namespace: params.Agent.Namespace,
 			},
-		}
+		})
+		toDelete = append(toDelete, &v1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: params.Agent.Namespace,
+			},
+		})
 	case spec.Deployment != nil:
 		reconciliationFunc = reconcileDeployment
-		toDelete = &v1.DaemonSet{
+		toDelete = append(toDelete, &v1.DaemonSet{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      name,
 				Namespace: params.Agent.Namespace,
 			},
-		}
+		})
+		toDelete = append(toDelete, &v1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: params.Agent.Namespace,
+			},
+		})
+	case spec.StatefulSet != nil:
+		reconciliationFunc = reconcileStatefulSet
+		toDelete = append(toDelete, &v1.DaemonSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: params.Agent.Namespace,
+			},
+		})
+		toDelete = append(toDelete, &v1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: params.Agent.Namespace,
+			},
+		})
```

Comment:
nit: `append`  is a variadic function
```suggestion
		toDelete = append(toDelete,
			&v1.DaemonSet{
				ObjectMeta: metav1.ObjectMeta{
					Name:      name,
					Namespace: params.Agent.Namespace,
				},
			},
			&v1.Deployment{
				ObjectMeta: metav1.ObjectMeta{
					Name:      name,
					Namespace: params.Agent.Namespace,
				},
			},
		)
```

Reply:
oh nice catch! sure this makes 100% sense and I can't tell how I missed it 😃